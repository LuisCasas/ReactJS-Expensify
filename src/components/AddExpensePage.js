import React from 'react';

const AddExpensePage = () => (
    <div>
        Add a Help page.
    </div>
);

export default AddExpensePage;