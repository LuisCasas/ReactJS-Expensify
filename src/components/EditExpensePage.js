import React from 'react';

const EditExpensePage = (props) => {

    console.log(props);

    return (
    <div>
       Amending the expense of {props.match.params.id}
    </div>
    );
};

export default EditExpensePage;