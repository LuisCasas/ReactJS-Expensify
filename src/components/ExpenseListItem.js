import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseItem = ({id, description, amount, createdAt, dispatch}) => (
    <div>
        <Link to={`/edit/${id}`} >Edit</Link>
        <p>Description: {description}</p>
        <p>Amount: {amount} - Created at: {createdAt}</p>
    </div>
);

export default ExpenseItem;