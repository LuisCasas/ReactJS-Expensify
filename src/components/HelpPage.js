import React from 'react';

const HelpPage = () => (
    <div>
        Add a Help page.
    </div>
);

export default HelpPage;