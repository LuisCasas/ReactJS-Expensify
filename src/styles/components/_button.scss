.button{
    background: $blue;
    border: none;
    color: #ffffff;
    padding: $m-size;
    font-weight: 300;
    font-size: $font-size-large;    
}