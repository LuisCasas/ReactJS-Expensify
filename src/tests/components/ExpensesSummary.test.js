import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render component ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={200} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render component ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={100} />);
    expect(wrapper).toMatchSnapshot();
});