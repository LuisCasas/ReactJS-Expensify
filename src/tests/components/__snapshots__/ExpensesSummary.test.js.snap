// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     with a total of 
    $2.00
  </h1>
</div>
`;

exports[`Should render component ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     with a total of 
    $1.00
  </h1>
</div>
`;
